metals/src/main/scala/scala/meta/internal/metals/codelenses/RunTestCodeLens.scala

```diff
@@ -56,25 +56,33 @@ final class RunTestCodeLens(
   ): Seq[l.CodeLens] = {
     val textDocument = textDocumentWithPath.textDocument
     val path = textDocumentWithPath.filePath
-    if (path.isAmmoniteScript || path.isWorksheet) {
-      Seq.empty
-    } else {
-      val distance = buffers.tokenEditDistance(path, textDocument.text, trees)
-      val lenses = for {
-        buildTargetId <- buildTargets.inverseSources(path)
-        buildTarget <- buildTargets.info(buildTargetId)
-        connection <- buildTargets.buildServerOf(buildTargetId)
-        // although hasDebug is already available in BSP capabilities
-        // see https://github.com/build-server-protocol/build-server-protocol/pull/161
-        // most of the bsp servers such as bloop and sbt might not support it.
-        if buildTarget.getCapabilities.getCanDebug || connection.isBloop || connection.isSbt,
-      } yield {
-        val classes = buildTargetClasses.classesOf(buildTargetId)
-        codeLenses(textDocument, buildTargetId, classes, distance, path)
-      }
-
-      lenses.getOrElse(Seq.empty)
+    val distance = buffers.tokenEditDistance(path, textDocument.text, trees)
+    val lenses = for {
+      buildTargetId <- buildTargets.inverseSources(path)
+      buildTarget <- buildTargets.info(buildTargetId)
+      connection <- buildTargets.buildServerOf(buildTargetId)
+      // although hasDebug is already available in BSP capabilities
+      // see https://github.com/build-server-protocol/build-server-protocol/pull/161
+      // most of the bsp servers such as bloop and sbt might not support it.
+      if buildTarget.getCapabilities.getCanDebug || connection.isBloop || connection.isSbt
+      if connection.isScalaCLI || !path.isAmmoniteScript || !path.isWorksheet
+    } yield {
+      val classes = buildTargetClasses.classesOf(buildTargetId)
+      val scalaCliScripts =
+        if (connection.isScalaCLI && path.isScalaScript)
+          scalaCliCodeLenses(textDocument, buildTargetId, classes)
+        else Seq.empty
+      val lenses = codeLenses(
+        textDocument,
+        buildTargetId,
+        classes,
+        distance,
+        path,
+      )
+      lenses ++ scalaCliScripts
```

Comment:
Is there a case when both `lenses` and `scalaCliScripts` will be nonempty? Aren't they mutually exclusive?

Reply:
I think that they are mutually exclusive but I have no proof. For example, for the example below, only one lens code is generated:
<img width="377" alt="Screenshot 2022-11-04 at 14 21 56" src="https://user-images.githubusercontent.com/46607934/199982708-edf1db29-2ae9-4800-b99e-d20f58b10b8a.png">

The `lenses` list contains the lenses that are generated from the symbol from the source, in the example above it is` test.sc` and the `codeLenses` function doesn't find any main class in `test.sc`. So I added a special case for Scala CLI.

Reply:
I think they're mutually exlusive, but we can wait for an expert opinion, @tgodzik

Reply:
lenses are created from the list of main classes as defined by the server.  I think the script main class is not found in the source so  it's not shown.